x11-packages/luanti/build.sh

```diff
@@ -0,0 +1,20 @@
+TERMUX_PKG_HOMEPAGE=https://www.luanti.org
+TERMUX_PKG_DESCRIPTION="An open source voxel game engine."
+TERMUX_PKG_LICENSE="LGPL-2.1"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=1:5.10.0
+TERMUX_PKG_REVISION=4
+TERMUX_PKG_SRCURL=https://github.com/minetest/minetest/archive/refs/tags/${TERMUX_PKG_VERSION:2}.zip
+TERMUX_PKG_SHA256=e74e994c0f1b188d60969477f553ad83b8ce20ee1e0e2dcd068120189cb0f56c
+TERMUX_PKG_AUTO_UPDATE=true
+TERMUX_PKG_DEPENDS="freetype, jsoncpp, libandroid-spawn, libc++, libcurl, libglvnd-dev, libgmp, libjpeg-turbo, libiconv, libpng, libsqlite, libvorbis, libxi, luajit, luanti-common, openal-soft, xdg-utils, zstd"
```

Comment:
```suggestion
TERMUX_PKG_DEPENDS="freetype, jsoncpp, libandroid-spawn, libc++, libcurl, libgmp, libjpeg-turbo, libiconv, libluajit, libpng, libsqlite, libvorbis, libx11, libxi, luanti-common, openal-soft, opengl, xdg-utils, zlib, zstd"
```

Reply:
Thank you for the suggestion, I agree that for some of these it is difficult to tell what the correct dependencies to mark are because multiple have the same or similar effect (`libxi` depends on `libx11` so pulls it in, `zstd` depends on `zlib` so pulls it in, etc.) so I will try to go with your suggestions. 

The reason I had written `libglvnd-dev` instead of `opengl` is because if I write `pkg install opengl` on Termux, the necessary file `/data/data/com.termux/files/usr/include/GLES2/gl2.h` does not get installed. For some reason, though, when `opengl` is placed in a dependency list for `./build-package.sh` (not `pkg install`), `./build-package.sh` automatically installs `libglvnd-dev`, but I don't understand how it is able to because `libglvnd-dev` is not in the `TERMUX_PKG_DEPENDS`  or `TERMUX_PKG_BUILD_DEPENDS` lines of `mesa` or `libglvnd`. 

Do you happen to know how that is able to work?

Reply:
`libglvnd-dev` is a subpackage of `libglvnd`. The package and it's subpackages are downloaded and extracted when using `-I`. AFAIK `TERMUX_PKG_ANTI_BUILD_DEPENDS` can be used to block them from pulling in.

Reply:
Thank you for explaining, that helped me to understand how it works and find the place where the dependency is included, I have found what you describe now, "`for subpkg in self.subpkgs`... `self.deps.add(subpkg.name)`" in line 155 of `buildorder.py`.

https://github.com/termux/termux-packages/blob/eee8df6a96933e6d4d6b4fb9b97ecde47490a477/scripts/buildorder.py#L155

Putting a print statement in there that just prints out the variables `self.name` and `subpkg.name` while `buildorder.py` runs shows a confirmation of this,

```
[buildorder.py]: subpackage dependency of packge libglvnd included: libglvnd-dev
```

This must be why `libglvnd-dev`  gets installed by `build-package.sh` when building packages that depend on the `opengl` metapackage or the `libglvnd` package directly. I just wanted to make a note of what is going on here that makes the `opengl` metapackage work as a dependency instead of `libglvnd-dev`, so that if this `buildorder.py` behavior ever changes in the future and affects the dependencies of `luanti`, there would be a record of why the `opengl` dependency was used at this time.